File: backend/BookingSystem.API/Models/Booking.cs
using BookingSystem.API.Enums;

namespace BookingSystem.API.Models;

public class Booking
{
    public int Id { get; set; }
    public int RoomId { get; set; }
    public required string CustomerName { get; set; }
    public required string CustomerEmail { get; set; }
    public string CustomerPhone { get; set; } = string.Empty;
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public DateOnly BookingDate { get; set; }
    public BookingStatus Status { get; set; }
}

File: frontend/src/api/booking.ts
import { baseAddress } from "./client";

export interface bookingRequest {
  roomId: number;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  startTime: string;
  endTime: string;
  bookingDate: string;
}

export default async function postBooking(
  booking: bookingRequest,
): Promise<bookingRequest> {
  const route = "/api/bookings/";
  const response = await fetch(baseAddress + route, {
    method: "POST",
    body: JSON.stringify(booking),
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  return data;
}
